cmd/empire/server.go

```diff
@@ -108,5 +108,16 @@ func newAuthenticator(c *cli.Context, e *empire.Empire) auth.Authenticator {
 		)
 	}
 
+	// After the user is authenticated, check their GitHub Team membership.
+	if teamId := c.String(FlagGithubTeam); teamId != "" {
+		authorizer := githubauth.NewTeamAuthorizer(client)
+		authorizer.TeamId = teamId
+
+		log.Println("Adding GitHub Team authorizer with the following configuration:")
+		log.Println(fmt.Sprintf("  Team ID: %v ", teamId))
+
+		return auth.WithAuthorization(authenticator, authorizer)
```

Comment:
Should we cache this authorization, like we do with the [org auth](https://github.com/mhahn/empire/blob/github-team-auth/cmd/empire/server.go#L107)? Without caching, every empire API hit incurs a full 1-2 round trips to GitHub, which makes everything slow. An in memory ~30 minute cache has worked pretty well.